tests/rptest/tests/cloud_retention_test.py

```diff
@@ -8,60 +8,84 @@
 # by the Apache License, Version 2.0
 
 from ducktape.utils.util import wait_until
+from ducktape.mark import matrix
 from rptest.services.cluster import cluster
 from rptest.tests.prealloc_nodes import PreallocNodesTest
 from rptest.clients.types import TopicSpec
 from rptest.clients.rpk import RpkTool
 from rptest.services.redpanda import SISettings, MetricsEndpoint
 from rptest.services.kgo_verifier_services import (
     KgoVerifierConsumerGroupConsumer, KgoVerifierProducer)
+from rptest.utils.mode_checks import skip_debug_mode
 
 
 class CloudRetentionTest(PreallocNodesTest):
-    segment_size = 1024 * 1024
     default_retention_segments = 2
     topic_name = "si_test_topic"
 
     def __init__(self, test_context):
+        super(CloudRetentionTest, self).__init__(test_context=test_context,
+                                                 node_prealloc_count=1,
+                                                 num_brokers=3)
+
+    def setUp(self):
+        # Defer cluster startup so we can tweak configs based on the type of
+        # test run.
+        pass
+
+    @cluster(num_nodes=4)
+    @matrix(max_consume_rate_mb=[20, None])
+    @skip_debug_mode
+    def test_cloud_retention(self, max_consume_rate_mb):
+        """
+        Test cloud retention with an ongoing workload.
+        """
+        if self.redpanda.dedicated_nodes:
+            num_partitions = 10
+            msg_size = 128 * 1024
+            msg_count = int(100 * 1024 * 1024 * 1024 / msg_size)
+            segment_size = 10 * 1024 * 1024
+        else:
+            num_partitions = 1
+            msg_size = 1024
+            msg_count = int(1024 * 1024 * 1024 / msg_size)
+            segment_size = 1024 * 1024
+
+        si_settings = SISettings(log_segment_size=segment_size)
+        self.redpanda.set_si_settings(si_settings)
+
         extra_rp_conf = dict(retention_local_target_bytes_default=self.
-                             default_retention_segments * self.segment_size,
+                             default_retention_segments * segment_size,
                              log_segment_size_jitter_percent=5,
                              group_initial_rebalance_delay=300,
                              cloud_storage_segment_max_upload_interval_sec=60,
                              cloud_storage_housekeeping_interval_ms=10_000)
+        self.redpanda.add_extra_rp_conf(extra_rp_conf)
 
-        super(CloudRetentionTest, self).__init__(
-            test_context=test_context,
-            node_prealloc_count=1,
-            num_brokers=3,
-            si_settings=SISettings(log_segment_size=self.segment_size),
-            extra_rp_conf=extra_rp_conf)
+        self.redpanda.start()
+        self._create_initial_topics()
 
-    @cluster(num_nodes=4)
```

Comment:
This needs a rebase to pick up the ok_to_fail that's on the test on the tip of `dev` (and this PR should remove the ok_to_fail)

Reply:
Done